docs/migrate-from-sidecar-to-ambient/migration.adoc

```diff
@@ -0,0 +1,2114 @@
+** <<observability-considerations, 6.2 Observability Considerations>>
+** <<configure-kiali-for-ambient-mode, 6.3 Configure Kiali for Ambient Mode>>
+* <<additional-resources, 7. Additional Resources>>
+
+---
+
+[[overview-of-sidecar-to-ambient-migration]]
+== 1. Overview of Sidecar to Ambient Migration
+
+[[migration-strategy]]
+=== 1.1 Migration Strategy
+
+The migration from sidecar to ambient mode follows a structured, phased approach to ensure a smooth transition with minimal disruption. Key principles include:
+
+1. **Gradual Transition**: Both sidecar and ambient proxies operate simultaneously during migration
+2. **Service-by-Service**: Migration can occur incrementally, one service or namespace at a time
+3. **Policy Preservation**: All existing security and traffic policies are maintained throughout the process
+4. **Rollback Support**: Each step can be reversed if issues arise
+
+[[key-architectural-changes]]
+=== 1.2 Key Architectural Changes
+
+[cols="1,2,2"]
+|===
+|Component |Sidecar Mode |Ambient Mode
+
+|**L4 Processing**
+|Envoy sidecar per pod
+|ZTunnel node proxy (shared)
+
+|**L7 Processing**
+|Envoy sidecar per pod
+|Waypoint proxy per namespace/service
+
+|**mTLS Enforcement**
+|Sidecar proxy
+|ZTunnel proxy
+
+|**Policy Enforcement**
+|Sidecar proxy
+|ZTunnel (L4) + Waypoint (L7)
+|===
+
+---
+
+[[prerequisites]]
+== 2. Prerequisites
+
+For this migration we are going to use the bookinfo application as an example. You can use any application that is currently running with sidecar injection enabled.
+
+[[technical-requirements]]
+=== 2.1 Technical Requirements
+
+Before starting migration, ensure your environment meets these requirements:
+
+For Kubernetes:
+
+- **Kubernetes 1.21+**: Minimum version for ambient mode support
+- **Sail Operator v1.27+**: Must be running Sail Operator version 1.27 or later
+- **Istio v1.27+**: Control plane must be v1.27 or later for ambient support
+- **Cluster Admin Access**: Required for namespace labeling and resource creation
+
+For Openshift:
+
+- **Openshift 4.19+**: Minimum version for ambient mode support
+- **Sail Operator v1.27+**: Must be running Sail Operator version 1.27 or later
+- **Istio v1.27+**: Control plane must be v1.27 or later for ambient support
+- **Cluster Admin Access**: Required for namespace labeling and resource creation
+
+[[cni-compatibility-requirements]]
+=== 2.1.1 CNI Compatibility Requirements
+
+**CNI Configuration Validation:**
+
+For Openshift clusters, the CNI must be `OVNKubernetes`. Other CNIs may work but are not officially supported. For Kubernetes, any CNI that supports NetworkPolicy should be compatible.
+
+[source,console]
+----
+# Verify CNI type compatibility
+kubectl get network.operator cluster -o jsonpath='{.spec.defaultNetwork.type}'
+# Expected: OVNKubernetes for OCP clusters
+
+# Check if multiple CNI plugins are configured
+kubectl get network.operator cluster -o yaml | grep -A 10 "additionalNetworks"
+----
```

Comment:
As this is a document for migrating from the sidecar mode to ambient mode and not about migrating the K8s cluster, I'm not sure if we need this section.